include/albatross/src/indexing/group_by.hpp

```diff
@@ -82,20 +82,32 @@ template <typename KeyType, typename ValueType> class GroupedBase {
 
   ValueType first_value() const { return map_.begin()->first; }
 
-  auto min_value() const {
+  // The min entry based on the values
+  auto min() const {
     const auto value_compare = [](const auto &x, const auto &y) {
       return x.second < y.second;
     };
-    return std::min_element(begin(), end(), value_compare)->second;
+    return *std::min_element(begin(), end(), value_compare);
```

Comment:
Do we need the `value_compare` method here? It looks like `min_element` might use `<` as a default operator.

Reply:
Yeah, without `value_compare` it'll try to compare the `std::pair<KeyType, ValueType>` but we want to only compare based on the value.